### PR TITLE
initial integration of tracing (previously tokio-trace) for logging model construction

### PR DESCRIPTION
**Pull request by salmans**

### CODE REVIEW

File: src/chase.rs [hunk 1]
```diff
@@ -49,7 +51,7 @@ pub struct Rel(pub String);
 
 impl Rel {
     pub fn new(name: &str) -> Rel {
-        Rel(name.to_string())
+        Rel(name.to_owned())
```
Comment (KiChjang) [inline src/chase.rs]: There used to be a performance difference between `to_owned` and `to_string`, but I've been told that there is no difference between the two nowadays.

File: src/chase/impl/basic.rs [hunk 2]
```diff
@@ -109,7 +111,7 @@ impl Model {
         let mut result = element;
         let mut element = Some(element);
         while element.is_some() {
-            let e= element.unwrap();
+            let e = element.unwrap();
```
Comment (KiChjang) [inline src/chase/impl/basic.rs]:
```rust
while let Some(e) = element {
    result = e;
    element = self.equality_history.get(e);
}
```

File: src/main.rs [hunk 3]
```diff
@@ -225,7 +228,12 @@ fn main() {
     }
 }
 
-fn process_solve(theory: &Theory, bound: Option<BoundCommand>, strategy: StrategyOption, count: Option<i32>, color: bool) {
+fn process_solve(theory: &Theory,
```
Comment (KiChjang) [inline src/main.rs]: Huh, I wonder why rust-fmt didn't complain about this formatting...

File: src/trace/subscriber.rs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+#[derive(Serialize, Deserialize)]
+pub struct ModelRecord {
+    event: Option<String>,
+    model_id: Option<u64>,
+    parent: Option<u64>,
+    model: Option<String>,
+}
+
+impl<'a> ModelRecord {
```
Comment (KiChjang) [inline src/trace/subscriber.rs]: I didn't see the usage of 'a in this struct?

File: src/trace/subscriber.rs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        ModelRecord {
+            event: None,
+            model_id: None,
+            parent: None,
+            model: None,
+        }
+    }
+}
+
+impl<'a> field::Visit for ModelRecord {
```
Comment (KiChjang) [inline src/trace/subscriber.rs]: Same question about 'a here.

File: src/chase.rs (hunk 1 above)
Comment (salmans) [reply src/chase.rs]: Good to know. I made the change based on what I read online but probably older material.

File: src/main.rs (hunk 3 above)
Comment (salmans) [reply src/main.rs]: No idea. What is the conventional way of breaking lines like this?

File: src/trace/subscriber.rs (hunk 4 above)
Comment (salmans) [reply src/trace/subscriber.rs]: ah nice catch- I went back and forth on the type of the struct but didn't cleanup properly.

File: src/main.rs (hunk 3 above)
Comment (KiChjang) [reply src/main.rs]:
```rust
fn process_solve(
    theory: &Theory,
    bound: Option<BoundCommand>,
    // ...
) {
```

**Pull request merged**